src/test/java/com/redhat/darcy/webdriver/WebDriverCookieManagerTest.java

```diff
@@ -0,0 +1,156 @@
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.redhat.darcy.webdriver;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.redhat.darcy.web.Cookie;
+import com.redhat.darcy.webdriver.internal.TargetedWebDriver;
+import com.redhat.darcy.webdriver.internal.WebDriverCookieManager;
+
+import org.junit.Test;
+import org.openqa.selenium.WebDriver;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.Date;
+
+public class WebDriverCookieManagerTest {
+
+    @Test
+    public void shouldProperlyAddACookieWithANameAndValue() {
+        Cookie cookie = new Cookie("chocolate", "chip");
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip");
+
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValueAndPath() {
+        Cookie cookie = new Cookie("chocolate", "chip", null);
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", null);
+
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValuePathAndExpiry() {
+        Cookie cookie = new Cookie("chocolate", "chip", "home", LocalDateTime.now());
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", "home", Date.from(Instant.now()));
```

Comment:
This probably isn't going to be a huge concern but, technically if for some reason the `Instant.now()` is long enough after `LocalDateTime.now()` above (not sure the precision of these two), these won't be equivalent. I would get an `Instant` _once_ above and then use that for both cookies.